packages/datagateway-search/src/table/investigationSearchTable.component.test.tsx

```diff
@@ -1,622 +1,527 @@
 
-  it('updates filter query params on date filter', () => {
-    applyDatePickerWorkaround();
+  it('should remove the selected row from cart if it is in the cart', async () => {
+    const addedCartItem: DownloadCartItem = {
+      entityId: 1,
+      entityType: 'investigation',
+      id: 1,
+      name: 'Test 1',
+      parentEntities: [],
+    };
 
-    const wrapper = createWrapper();
+    cartItems.push(addedCartItem);
 
-    const filterInput = wrapper.find(
-      'input[id="investigations.end_date filter to"]'
-    );
-    filterInput.instance().value = '2019-08-06';
-    filterInput.simulate('change');
+    renderComponent();
 
-    expect(history.length).toBe(2);
-    expect(history.location.search).toBe(
-      `?filters=${encodeURIComponent('{"endDate":{"endDate":"2019-08-06"}}')}`
-    );
+    // wait for data to finish loading
+    expect(await screen.findByText('Test title 1')).toBeInTheDocument();
+    expect(screen.getByText('Test name 1')).toBeInTheDocument();
 
-    filterInput.instance().value = '';
-    filterInput.simulate('change');
+    // pretend the server has removed the item from the cart
+    // create a new array to trigger useMemo update
+    cartItems = [];
 
-    expect(history.length).toBe(3);
-    expect(history.location.search).toBe('?');
+    // clicks on the row checkbox
+    await user.click(screen.getByRole('checkbox', { name: 'select row 0' }));
 
-    cleanupDatePickerWorkaround();
+    // the checkbox should be checked
+    expect(
+      await screen.findByRole('checkbox', { name: 'select row 0' })
+    ).not.toBeChecked();
   });
 
-  it('updates sort query params on sort', () => {
-    const wrapper = createWrapper();
-
-    wrapper
-      .find('[role="columnheader"] span[role="button"]')
-      .first()
-      .simulate('click');
+  it('selected rows only considers relevant cart items', async () => {
+    cartItems = [
+      {
+        entityId: 2,
+        entityType: 'investigation',
+        id: 1,
+        name: 'test',
+        parentEntities: [],
+      },
+      {
+        entityId: 1,
+        entityType: 'dataset',
+        id: 2,
+        name: 'test',
+        parentEntities: [],
+      },
+    ];
 
-    expect(history.length).toBe(2);
-    expect(history.location.search).toBe(
-      `?sort=${encodeURIComponent('{"title":"asc"}')}`
-    );
-  });
+    renderComponent();
 
-  it('calls addToCart mutate function on unchecked checkbox click', () => {
-    const addToCart = jest.fn();
-    (useAddToCart as jest.Mock).mockReturnValue({
-      mutate: addToCart,
-      loading: false,
+    const selectAllCheckbox = await screen.findByRole('checkbox', {
+      name: 'select all rows',
     });
-    const wrapper = createWrapper();
-
-    wrapper.find('[aria-label="select row 0"]').last().simulate('click');
 
-    expect(addToCart).toHaveBeenCalledWith([1]);
+    expect(selectAllCheckbox).not.toBeChecked();
+    expect(selectAllCheckbox).toHaveAttribute('data-indeterminate', 'false');
```

Comment:
As above with the earlier data-indeterminate comment

Reply:
Same comment as above.